File: java/merchants-guide-to-galaxy/src/main/java/com/codingkapoor/merchantsguidetogalaxy/util/UserInputFileReader.java
package com.codingkapoor.merchantsguidetogalaxy.util;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.net.URISyntaxException;
import java.net.URL;
import java.nio.charset.Charset;
import java.nio.file.Files;
import java.util.List;

public class UserInputFileReader {

	public List<String> read(String fileName) throws IOException, URISyntaxException {

		File file;

		if (fileName != null)
			file = new File(fileName);
		else {
			URL url = getClass().getClassLoader().getResource("default-user-input.txt");
			if (url == null)
				throw new FileNotFoundException();

			file = new File(url.toURI());
		}

		return Files.readAllLines(file.toPath(), Charset.defaultCharset());
	}
}

File: scala/page-selection-by-keyword-matching-I-X/src/main/scala/com/codingkapoor/pageselectionbykeyword/util/UserInputFileReader.scala
package com.codingkapoor.pageselectionbykeyword.util

import com.codingkapoor.pageselectionbykeyword.model.Page
import com.codingkapoor.pageselectionbykeyword.model.Query

trait UserInputFileReader {
  
  final val totalKeywordsAllowed = 8
  final val defaultUserInputFile = "/input.txt"
  
  def readUserInputFile(fileName: Option[String]): (List[Query], List[Page])
}
